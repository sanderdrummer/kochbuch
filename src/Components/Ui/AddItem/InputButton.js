import React from 'react';
import PropTypes from 'prop-types';

const InputButton = (props) => (
    <form onSubmit={(e) => {e.preventDefault();props.triggerOnClick()}} className="field has-addons">
        <div className="control grow">
            <input className="input"
                   required
                   autoFocus="true"
                   onChange={props.updateValue}
                   value={props.value}
                   type={props.inputType || 'text'}
                   placeholder={props.placeholder} />
        </div>
        <div className="control">
            <button onClick={props.triggerOnClick} type="button" className={
                props.isLoading ? 'button is-primary is-loading': 'button is-primary'
            }>
                {props.buttonLabel}
            </button>
        </div>
    </form>
);

InputButton.propTypes = {
    inputType: PropTypes.string,
    buttonLabel: PropTypes.string,
    placeholder: PropTypes.string,
    updateValue: PropTypes.func,
    triggerOnClick: PropTypes.func,
    isLoading: PropTypes.bool
};

export default InputButton;