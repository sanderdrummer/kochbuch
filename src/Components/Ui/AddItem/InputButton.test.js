import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import InputButton from './InputButton';
import React from 'react';
import toJson from 'enzyme-to-json';

configure({
    adapter: new Adapter()
});

describe('InputButton tests', () => {

    let wrapper;
    let triggerOnClick;

    beforeEach(() => {

        triggerOnClick = jest.fn();
        wrapper = shallow(<InputButton triggerOnClick={triggerOnClick}/>);
    });

    afterEach(() => {
        triggerOnClick.mockReset();    
    });

    it('should render an input and a button' , () => {

        expect(wrapper.find('button.button.is-primary').length).toBeTruthy();
        expect(wrapper.find('input.input').length).toBeTruthy();        
    });

    it('should be submitted on click', () => {
        const button = wrapper.find('button.button');
        button.simulate('click');

        expect(triggerOnClick.mock.calls.length).toEqual(1);
    });

    it('should render correctly', () => {

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});