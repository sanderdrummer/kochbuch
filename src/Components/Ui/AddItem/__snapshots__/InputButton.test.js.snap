// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputButton tests should render correctly 1`] = `
<form
  className="field has-addons"
  onSubmit={[Function]}
>
  <div
    className="control grow"
  >
    <input
      autoFocus="true"
      className="input"
      required={true}
      type="text"
    />
  </div>
  <div
    className="control"
  >
    <button
      className="button is-primary"
      onClick={[Function]}
      type="button"
    />
  </div>
</form>
`;
